src/core/command-executor.ts

```diff
@@ -1,67 +1,80 @@
-import { CommandMap } from './interfaces/command-map';
+import { CommandMapInstance } from './interfaces/command-map';
 import { ParsedArgs } from './interfaces/parsed-args';
 
 export class CommandExecutor {
+    private map: CommandMapInstance[];
 
     // TODO: remove boolean and make it throw instead
-    public execute(args: ParsedArgs, map: CommandMap[]): boolean {
-        const command = this.resolveCommand(map, args.name);
+    // There is no good reason to throw here, we just want to know if the command executed
+    public execute(args: ParsedArgs, map: CommandMapInstance[]): boolean {
+        this.map = map;
+        const command = this.resolveCommand(args.name);
         if (!command) {
             return false;
         }
 
         this.injectOptions(command, args);
-        command.instance.execute(...this.getArgMap(command, args));
+        const params = this.mapParameters(command, args);
+        command.instance.execute(...params);
 
         return true;
     }
 
-    private resolveCommand(map: CommandMap[], name: string) {
-        return this.resolveCommandWithName(map, name)
-            || this.resolveCommandWithAlias(map, name);
+    // Find a command in the array
```

Comment:
One big reason why I would like you to read `Clean Code`. If you were able to write this comment so easily it's for one reason.

It's clear what the method do here because the semantic makes sense (atleast to me and you). 
You create two problems by adding a comment here:
- It holds no value and add things to read -> longer comprehension time
- We now have to maintain the code and a comment, which if we forget, we'll have a lying comment which will mislead people who read the code

So instead of writing comments, just write clear and simple block of codes with well chosen names.  
By reading your comment here I think you'd have preferred `findCommand` which I prefer too !
Feel free to rename methods / constants / variables / classes if you feel uncomfortable with the name or find a better one.

Anyway this is discussed in the book I won't rewrite it completely, haha.